monai/application/datasets.py

```diff
@@ -120,3 +120,114 @@ def _generate_data_list(self, dataset_dir):
                 raise ValueError("section name can only be: training, validation or test.")
             data.append({"image": image_files_list[i], "label": image_class[i]})
         return data
+
+
+class DecathlonDataset(Randomizable, CacheDataset):
+    """
+    The Dataset to automatically download the data of Decathlon challenge (http://medicaldecathlon.com/)
+    and generate items for training, validation or test. It's based on `CacheDataset` to accelerate
+    the training process.
+
+    Args:
+        root_dir: target directory to download and load Decathlon dataset.
+        task: which task to download and execute: one of list ("Task01_BrainTumour", "Task02_Heart",
+            "Task03_Liver", "Task04_Hippocampus", "Task05_Prostate", "Task06_Lung", "Task07_Pancreas",
+            "Task08_HepaticVessel", "Task09_Spleen", "Task10_colon").
+        section: expected data section, can be: `training`, `validation` or `test`.
+        transform: transforms to execute operations on input data.
+        download: whether to download and extract the Decathlon from resource link, default is False.
+            if expected file already exists, skip downloading even set it to True.
+            user can manually copy tar file or dataset folder to the root directory.
+        seed: random seed to randomly split `training`, `validation` and `test` datasets, defaut is 0.
+        val_frac: percentage of of validation fraction from the `training` section, default is 0.2.
+            Decathlon data only contains `training` section with labels and `test` section without labels,
+            so randomly select fraction from the `training` section as the `validation` section.
+        cache_num: number of items to be cached. Default is `sys.maxsize`.
+            will take the minimum of (cache_num, data_length x cache_rate, data_length).
+        cache_rate: percentage of cached data in total, default is 1.0 (cache all).
+            will take the minimum of (cache_num, data_length x cache_rate, data_length).
+        num_workers: the number of worker threads to use.
+            If 0 a single thread will be used. Default is 0.
+
```

Comment:
could put a quick code example here, something like:
```
Examples::

       >>>  DecathlonDataset()
```

Reply:
added example code in the latest commit.
Thanks.